server/index.js

```diff
@@ -1,49 +1,34 @@
 const express = require("express");
-const expressWs = require("express-ws");
-
 const bodyParser = require("body-parser");
 const cookieParser = require("cookie-parser");
 
-const db = require("./postgresQueries").queries;
-
-const wsConnectHandler = require("./webSocket").connect;
+const expressWs = require("express-ws");
+const wsConnectHandler = require("./lib/webSocket").connect;
 
 const usersHandlers = require("./routes/users");
 const problemsHandlers = require("./routes/problems");
 const publicationsHandlers = require("./routes/publications");
-const OAuthFlowResponseHandlers = require("./routes/oauth-flow");
+const orcidAuthHandlers = require("./routes/auth/orcid");
 
 const fb = require("./feedback");
-const multer = require("multer");
-const MulterAzureStorage = require("multer-azure-storage");
 
-const blobService = require("./blobService.js");
+const blobService = require("./lib/blobService");
 const upload = blobService.upload;
 blobService.initialise();
 
 const app = express();
-const wss = expressWs(app).getWss();
+expressWs(app);
 
-const port = process.env.PORT || 3001;
-
-const noop = () => {};
+const pino = require('express-pino-logger')();
+app.use(pino);
 
-const wss_interval = setInterval(() => {

```

Comment:
Is the pinging removed as a part of replacing the current WebSocket API or is there another reason?

Reply:
Hey @MoritzLangenstein,

Actually I wasn't sure whether it had any purpose other than being a beacon to the clients.
Is there any particular reason for doing this?

Happy to put it back in, just not sure if there's a functional (maybe forcing keep-alive or sth?) or is it just a debug utility to ensure communication flow?

Reply:
As far as I understand it, the code ensures that client connections are terminated using the WebSocket ping messages (which are handled by the client browsers internally) if the connection was broken but not terminated in a normal way (one example I read about was unplugging the Internet connection) - I never looked into whether such situations actually occurred in more detail. So if you know that this is (now) handled by the protocol natively, you can remove the code - otherwise it doesn’t do any harm either.

Reply:
@MoritzLangenstein Agreed, makes sense to terminate abandoned sockets. Restored this section of code.

Generally wondering whether we need to push anything over web sockets here rather than just plain HTTP/HTTPS but will leave that intact right now as I'm not sure what the implications could be :)

Reply:
Right now we only use websocket to subscribe to data changes and send notifications that the data has changed (not what it has changed to, we use the normal restful HTTPS API requests for that). If you can find a middleware that could automate this connection between the DB and the UI, it would replace this entire system on the server and on the client API proxy class.

Reply:
Ah perfect, that's what I thought as notifications are a very common use case for sockets.
Wasn't sure if it was a requirement of one of the dependencies so didn't want to take it out completely.

I think for the sake of simplicity clients polling every 10 seconds should do a good enough job with the same task at hand (basically one less connection type to maintain in the UI)

Reply:
With the current design polling would be very expensive in most parts, though. Especially the more complicated problem page took about 60 network requests to load all content which by itself took some time but could at least be cached - but even currently some “events” make all requests at least hit the cache all at once while rebuilding the problem page data structure which produces a noticeable lag for the user. With the current problem page, interactivity and simplicity of network don’t go together (maybe the server could build the structure internally and only send versioned diffs) but while possible in theory on other pages, polling wouldn’t work on our main one. A replacement would therefore only make the code on the frontend and backend cleaner and less error-prone but not change that live updates need some sort of second continuous bidirectional connection.

Reply:
Thanks @MoritzLangenstein, all valid points. Didn't realise it'd trigger 60 requests as I didn't get to a problem page yet. Sounds like socket might be the way to go for the current data flow.

I think it's a good idea to explore caching further. If we begin to get concerned about general latency we can always drop an external cache in there (Redis or sth). That could cache either individual DB queries or full precomputed data objects needed to render a problem.
I'll need to get more data first but it's an easy enough addition :)